src/interactions/interaction.ts

```diff
@@ -11,8 +11,53 @@ module Plottable {
      */
     protected _componentToListenTo: Component;
 
-    public _anchor(component: Component) {
+    private _anchorCallback = (component: Component) => this._anchor(component);
+
+    private _isAnchored: Boolean;
+
+    /* tslint:disable: */
+    // tslint disable does not really work
+    protected _anchor(component: Component) {
+      this._isAnchored = true;
+    }
+
+    protected _unanchor() {
+      this._isAnchored = false;
+    }
+    /* tslint:enable */
+
+    /**
+     * Attaches current interaction to a Component. If the interaction was already
+     * attached to a Component, it first detaches itself from the old component.
+     *
+     * @param {Component} component The component to which to attach the interaction.
+     *
+     * @return {Interaction}
+     */
+    public attachTo(component: Component) {
+      if (this._componentToListenTo) {
+        this.detachFrom(this._componentToListenTo);
+      }
```

Comment:
Possibly a premature optimization but should this no-op  if
component === this._componentToListenTo